vm/interpreter/src/circ_supply.rs

```diff
@@ -0,0 +1,254 @@
+                ExitCode::ErrIllegalState,
+                "failed to get reward actor for cumputing total supply",
+            )
+        })?
+        .ok_or_else(|| actor_error!(ErrIllegalState; "Actor address ({}) does not exist", addr))
+}
+
+pub fn get_fil_vested<DB: BlockStore>(
+    pre_ignition: &GenesisInfo,
+    post_ignition: &GenesisInfo,
+    height: ChainEpoch,
+    state_tree: &StateTree<DB>,
+) -> Result<TokenAmount, ActorError> {
+    let mut return_value = TokenAmount::default();
+
+    if height <= UPGRADE_IGNITION_HEIGHT {
+        for actor in &pre_ignition.genesis_msigs {
+            return_value += &actor.initial_balance - actor.amount_locked(height);
+        }
+    } else {
+        for actor in &post_ignition.genesis_msigs {
+            return_value +=
+                &actor.initial_balance - actor.amount_locked(height - actor.start_epoch);
+        }
+    }
+
+    for actor in &pre_ignition.genesis_actors {
+        let state = get_actor_state(state_tree, &actor.addr)?;
+        let diff = &actor.init_bal - state.balance;
+        if diff > TokenAmount::default() {
+            return_value += diff
+        }
+    }
+
+    if height < UPGRADE_ACTORS_V2_HEIGHT {
+        return_value += &pre_ignition.genesis_pledge + &pre_ignition.genesis_market_funds;
+    }
+
+    Ok(return_value)
+}
+
+pub fn get_fil_mined<DB: BlockStore>(
+    state_tree: &StateTree<DB>,
+) -> Result<TokenAmount, Box<dyn StdError>> {
+    let reward_actor = get_actor_state(state_tree, &*REWARD_ACTOR_ADDR)?;
+    let reward_state: reward::State = state_tree
+        .store()
+        .get(&reward_actor.state)?
+        .ok_or_else(|| "Failed to get Rewrad Actor State".to_string())?;
+
+    Ok(reward_state.total_storage_power_reward().clone())
+}
+
+pub fn get_fil_market_locked<DB: BlockStore>(
+    state_tree: &StateTree<DB>,
+) -> Result<TokenAmount, Box<dyn StdError>> {
+    let market_actor = get_actor_state(state_tree, &*STORAGE_MARKET_ACTOR_ADDR)?;
+
+    let market_state: market::State = state_tree
+        .store()
+        .get(&market_actor.state)?
+        .ok_or_else(|| "Failed to get Market Actor State".to_string())?;
+
+    Ok(market_state.total_locked())
+}
+
+pub fn get_fil_power_locked<DB: BlockStore>(
+    state_tree: &StateTree<DB>,
+) -> Result<TokenAmount, Box<dyn StdError>> {
+    let power_actor = get_actor_state(state_tree, &*STORAGE_POWER_ACTOR_ADDR)?;
+
+    let power_state: power::State = state_tree
+        .store()
+        .get(&power_actor.state)?
+        .ok_or_else(|| "Failed to get Power Actor State".to_string())?;
+
+    Ok(power_state.total_locked().clone())
+}
+
+pub fn get_fil_reserve_disbursed<DB: BlockStore>(
+    state_tree: &StateTree<DB>,
+) -> Result<TokenAmount, Box<dyn StdError>> {
+    let power_actor = get_actor_state(state_tree, &RESERVE_ADDRESS)?;
+
+    // If money enters the reserve actor, this could lead to a negative term
+    Ok(FIL_RESERVED.clone() - power_actor.balance)
```

Comment:
```suggestion
    let reserve_actor = get_actor_state(state_tree, &RESERVE_ADDRESS)?;

    // If money enters the reserve actor, this could lead to a negative term
    Ok(&*FIL_RESERVED - reserve_actor.balance)
```